code/game/objects/items/storage/storage.dm

```diff
@@ -722,6 +722,10 @@ W is always an item. stop_warning prevents messaging. user may be null.**/
 	return TRUE
 
 /obj/item/storage/proc/dump_into(obj/item/storage/M, mob/user)
+	if(storage_holster == TRUE && contents.len >= (storage_slots-1))
```

Comment:
This doesn't test if whether a gun is already holstered, and you'd only be able to check for this in either a `/obj/item/storage/belt/gun` or a `/obj/item/storage/belt/gun/flaregun` because it is `/obj/item/storage/belt/gun` that defines `var/list/holstered_guns`. In otherwords, currently if you have a gun holstered, then this proc will still leave an empty space for a gun that is already present.

Reply:
i see didn't think of that case. thanks that will avoid use to do another fix in the future...
need to activate my brain  for this.